sklearn/preprocessing/_function_transformer.py

```diff
@@ -117,6 +117,7 @@ def fit(self, X, y=None):
         if (self.check_inverse and not (self.func is None or
                                         self.inverse_func is None)):
             self._check_inverse_transform(X)
+        self.n_features_out_ = None
```

Comment:
should we set the `n_features_out_` from the transform on the given `X` if `validate=True`?

Reply:
you mean if ``check_inverse`` both are present? Otherwise we never call ``transform`` on X.

Reply:
yes. But also, the docstring says is validate is true, we check X. I was thinking we could expand that definition to mean we check X, and we remember what the output of transform is supposed to look like.

Reply:
That seems like a very costly operation just for the purpose of computing the number of features? It seems pretty unexpected to me.